src/dbnode/storage/bootstrap/bootstrapper/fs/source.go

```diff
@@ -982,12 +989,22 @@ func (s *fileSystemSource) bootstrapFromIndexPersistedBlocks(
 			continue
 		}
 
+		fsOpts := s.fsopts
+		verify := s.opts.IndexSegmentsVerify()
+		if verify {
+			// Make sure for this call to read index segments
+			// to validate the index segment.
+			// If fails validation will rebuild since missing from
+			// fulfilled range.
+			fsOpts = fsOpts.SetIndexReaderAutovalidateIndexSegments(true)
```

Comment:
+1  on this change, are there other places that we'd want to also validate index segments on read or are we already doing it?

Reply:
Self f/u on this: It looks like the only other place we are reading in index segments is in the persist manager after flush the segments to disk. We've just written to disk at that point so that should be fine.